crates/stackable-operator/src/client.rs

```diff
@@ -626,38 +633,45 @@ where
     }
 }
 
-pub async fn initialize_operator(field_manager: Option<String>) -> Result<Client> {
-    let _ = KUBERNETES_CLUSTER_DOMAIN
-        .set(retrieve_cluster_domain().context(ResolveKubernetesClusterDomainSnafu)?);
-    create_client(field_manager).await
-}
-
-async fn create_client(field_manager: Option<String>) -> Result<Client> {
+pub async fn initialize_operator(
+    cli_kubernetes_cluster_domain: &Option<DomainName>,
```

Comment:
Likewise, this could then take/forward the opts type, instead of having to know every involved field. That should also be more explicit in the calling code than a list of `None,None,None,None,None,`

Reply:
Please have a look at ce51bc5940cd63bc4047afc3955b73a80030156a